components/scream/src/share/tests/pack_tests.cpp

```diff
@@ -1,11 +1,283 @@
+    vector_novec for (int i = 0; i < Pack::n; ++i)  \
+      dc[i] = impl(a[i], c);                        \
+    compare_packs(dc, d);                           \
+    d = op(c, b);                                   \
+    vector_novec for (int i = 0; i < Pack::n; ++i)  \
+      dc[i] = impl(c, b[i]);                        \
+    compare_packs(dc, d);                           \
+  } while (0)
+
+#define test_pack_gen_unary_fn(op, impl) do {     \
+  Pack a, b, ac;                                  \
+  scalar c;                                       \
+  setup(a, b, c, true);                           \
+  a = op(abs(b));                                 \
+  vector_novec for (int i = 0; i < Pack::n; ++i)  \
+    ac[i] = impl(std::abs(b[i]));                 \
+  compare_packs(ac, a);                           \
+} while (0)
+
+#define test_pack_gen_unary_stdfn(op)           \
+  test_pack_gen_unary_fn(op, std::op)
+
+#define test_mask_gen_bin_op_all(op) do {           \
+    Pack a, b;                                      \
+    scalar c;                                       \
+    Mask m, mc;                                     \
+    setup(a, b, c);                                 \
+    m = a op b;                                     \
+    vector_novec for (int i = 0; i < Pack::n; ++i)  \
+      mc.set(i, a[i] op b[i]);                      \
+    compare_masks(mc, m);                           \
+    m = a op c;                                     \
+    vector_novec for (int i = 0; i < Pack::n; ++i)  \
+      mc.set(i, a[i] op c);                         \
+    compare_masks(mc, m);                           \
+    m = c op b;                                     \
+    vector_novec for (int i = 0; i < Pack::n; ++i)  \
+      mc.set(i, c op b[i]);                         \
+    compare_masks(mc, m);                           \
+  } while (0)
+
+  static void run () {
+    test_pack_gen_assign_op_all(=);
+    test_pack_gen_assign_op_all(+=);
+    test_pack_gen_assign_op_all(-=);
+    test_pack_gen_assign_op_all(*=);
+    test_pack_gen_assign_op_all(/=);
+
+    test_pack_gen_bin_op_all(+);
+    test_pack_gen_bin_op_all(-);
+    test_pack_gen_bin_op_all(*);
+    test_pack_gen_bin_op_all(/);
+
+    test_pack_gen_bin_fn_all(min, scream::util::min);
+    test_pack_gen_bin_fn_all(max, scream::util::max);
+
+    test_pack_gen_unary_stdfn(abs);
+    test_pack_gen_unary_stdfn(exp);
+    test_pack_gen_unary_stdfn(log);
+    test_pack_gen_unary_stdfn(log10);
+    test_pack_gen_unary_stdfn(tgamma);
+
+    test_mask_gen_bin_op_all(==);
+    test_mask_gen_bin_op_all(>=);
+    test_mask_gen_bin_op_all(<=);
+    test_mask_gen_bin_op_all(>);
+    test_mask_gen_bin_op_all(<);
+
+    test_conversion();
+    test_unary_min_max();
+    test_range();
+  }
+};
+
+template <typename Scalar, int PACKN>
+const double TestPack<Scalar,PACKN>::tol =
+  2*std::numeric_limits<Scalar>::epsilon();
+
+TEST_CASE("Pack", "scream::pack") {
+  TestPack<int,16>::run();
```

Comment:
Since we're testing just a small set, use SCREAM_PACK_SIZE here instead of the explicit 16.